dll/cpl/powercfg/advanced.c

```diff
@@ -308,11 +308,9 @@ Adv_InitDialog(
 
     GetPwrCapabilities(&spc);
 
-    if (spc.SystemS1 || spc.SystemS2 || spc.SystemS3)
-        bSuspend=TRUE;
-
-    bHibernate = spc.HiberFilePresent;
-    bShutdown = spc.SystemS5;
+    bSuspend = IS_PWR_SUSPEND_ALLOWED(&spc);
+    bHibernate = IS_PWR_HIBERNATE_ALLOWED(&spc);
```

Comment:
`SystemS4` was not checked here before, was that an omission?

Reply:
An omission. On that page, it's not wishable to add in the lists the "hibernate" option if the SystemS4 state (prerequisite for supporting hibernation) isn't supported. I'll make a screenshot to show which page I'm talking about.